tests/datasets/test_geo.py

```diff
@@ -152,6 +154,23 @@ def test_and_nongeo(self, dataset: GeoDataset) -> None:
         ):
             dataset & ds2  # type: ignore[operator]
 
+    def test_files_property_for_non_existing_file_or_dir(self, tmp_path: Path) -> None:
+        paths = [tmp_path.as_posix(), (tmp_path / "non_existing_file.tif").as_posix()]
```

Comment:
```suggestion
        paths = [str(tmp_path), str(tmp_path / "non_existing_file.tif")]
```
We're working on adding pathlib support in a different PR. For now, let's just cast to a string. We also need to test on Windows which isn't POSIX.